platform/lib/schema/index.ts

```diff
@@ -436,6 +436,65 @@ export class ObjectSetting<P extends Props> extends Setting<
   }
 }
 
+function isMap<K, V>(o: any): o is Map<K, V> {
+  return Object.prototype.toString.call(o) === '[object Map]';
+}
+
+// We need this helper for the types to be correct
+// (otherwise it assumes an array of A|B instead of a tuple [A,B])
+const toTuple = <A, B>(a: A, b: B): [A, B] => [a, b];
+
+type MapOfOptions<K, V> = SettingOptions<Map<K, V>>;
+
+export class MapOfSetting<K, V> extends Setting<Map<K, V>> {
+  constructor(
+    private readonly keySetting: Setting<K>,
+    private readonly valueSetting: Setting<V>,
+    options: MapOfOptions<K, V> = {}
+  ) {
+    super(options);
+  }
+
+  process(obj: any, context?: string): Map<K, V> {
+    if (isPlainObject(obj)) {
+      const res = Object.keys(obj).map(key => {
+        const validatedKey = this.keySetting.validate(
+          key,
+          toContext(context, key)
+        );
+        const validatedValue = this.valueSetting.validate(
+          obj[key],
+          toContext(context, key)
+        );
+        return toTuple(validatedKey, validatedValue);
```

Comment:
Yeah, I did that at first, but added `toTuple` to have a place to add a comment. But I agree, I'll change it. :+1: